patches/core/ungoogled-chromium/fix-building-without-enabling-reporting.patch

```diff
@@ -22,3 +22,157 @@
        "net/reporting_service_proxy.cc",
        "net/reporting_service_proxy.h",
      ]
+--- a/third_party/blink/renderer/core/frame/local_frame.cc
++++ b/third_party/blink/renderer/core/frame/local_frame.cc
+@@ -38,6 +38,7 @@
+ #include "base/unguessable_token.h"
+ #include "mojo/public/cpp/bindings/self_owned_receiver.h"
+ #include "mojo/public/cpp/system/message_pipe.h"
++#include "net/net_buildflags.h"
+ #include "services/data_decoder/public/mojom/resource_snapshot_for_web_bundle.mojom-blink.h"
+ #include "services/network/public/cpp/features.h"
+ #include "services/network/public/mojom/content_security_policy.mojom-blink.h"
+@@ -551,7 +552,9 @@ void LocalFrame::Trace(Visitor* visitor)
+   visitor->Trace(raw_system_clipboard_);
+   visitor->Trace(virtual_keyboard_overlay_changed_observers_);
+   visitor->Trace(pause_handle_receivers_);
++#if BUILDFLAG(ENABLE_REPORTING)
+   visitor->Trace(reporting_service_);
++#endif
+ #if defined(OS_MAC)
+   visitor->Trace(text_input_host_);
+ #endif
+@@ -2273,6 +2276,7 @@ const base::UnguessableToken& LocalFrame
+   return base::UnguessableToken::Null();
+ }
+ 
++#if BUILDFLAG(ENABLE_REPORTING)
+ mojom::blink::ReportingServiceProxy* LocalFrame::GetReportingService() {
+   if (!reporting_service_.is_bound()) {
+     GetBrowserInterfaceBroker().GetInterface(
+@@ -2281,6 +2285,7 @@ mojom::blink::ReportingServiceProxy* Loc
+   }
+   return reporting_service_.get();
+ }
++#endif
+ 
+ // static
+ void LocalFrame::NotifyUserActivation(
+--- a/third_party/blink/renderer/core/frame/local_frame.h
++++ b/third_party/blink/renderer/core/frame/local_frame.h
```

Comment:
@Ahrotahn's fix also removes the `reporting_service_` field of `LocalFrame`, as shown in the snippet below:

```
@@ -903,9 +906,11 @@
   // const methods.
   //
   // LocalFrame can be reused by multiple ExecutionContext.
+#if BUILDFLAG(ENABLE_REPORTING)
   mutable HeapMojoRemote<mojom::blink::ReportingServiceProxy,
                          HeapMojoWrapperMode::kWithoutContextObserver>
       reporting_service_{nullptr};
+#endif
```

While this is likely not *necessary*, I feel like it makes sense; if upstream adds more code that uses the `reporting_service` and we have disabled the field, we will get a compiler error instead of a runtime error.

Reply:
I have added this to the patch.